### DISCUSSION TITLE
Checkboxes always Checked?

### DISCUSSION THREAD

**Discussion by kevjustice** [Budibase/budibase#6869] [Help]
I'm a newbie to Budibase.  I have a mysql data source with a simple clients table that has a field "Active" which is a bit field.  I had Budibase auto-generate the CRUD for me however the checkbox is always checked regardless of what the value in the table is.

For the update (and add new), I fixed it by adding this javascript to the save button for the active field:
if($("Form.Fields.active") == true) {
	return true;
} else {
	return false;
}
And now the value saves correctly (although I'm not sure why this would fix it).  However the value when going into the edit form or on the main form is always checked.  

I added a text field to show the actual data (to the right of "edit" link so you can see the value.  Please advise what I did wrong.

![image](https://user-images.githubusercontent.com/7374411/180605539-329d56b8-f7f1-4c30-afaa-68666bfd68f8.png)

**Comment by melohagan** [comment]
Hey @kevjustice 

My guess is that within Budibase, the 'Active' column is *Text* type. This means "0" and "1" are both truthy true. 

If you edit the 'Active' column in Budibase are you able to change the type to *Boolean* ?
![Screenshot 2022-07-25 at 14 56 49](https://user-images.githubusercontent.com/101575380/180794790-077c1fd4-b0a8-4994-9a83-6085b5c04d9a.png)

If not, you may need to change the data type within your MySQL table, or you could use a <a href="https://docs.budibase.com/docs/formula">Formula</a> column that, similar to the code you posted, returns a boolean.
